### PR TITLE
build: fixups

### PR DESCRIPTION
**Pull request by gdams**
1st fix: removes error fatal: No tags can describe ''.
2nd fix: renames j2sdk-image to be the tag name making it clearer once
extracted
3rd fix: remove case sensitive dependancy for checking repo name
4th fix: shallow clone the `git fetch --tags` command

### CODE REVIEW

File: makejdk.sh [hunk 1]
```diff
@@ -215,9 +215,10 @@ getOpenJDKUpdateAndBuildVersion()
     # It does exist and it's a repo other than the AdoptOpenJDK one
     cd "${WORKING_DIR}/${OPENJDK_REPO_NAME}" || return
     echo "${git}Pulling latest tags and getting the latest update version"
-    git fetch --tags
-    OPENJDK_UPDATE_VERSION=$(git describe --abbrev=0 --tags | cut -d'u' -f 2 | cut -d'-' -f 1)
-    OPENJDK_BUILD_NUMBER=$(git describe --abbrev=0 --tags | cut -d'b' -f 2 | cut -d'-' -f 1)
+    git fetch -q --tags "${GIT_CLONE_ARGUMENTS[@]}"
```
Comment (neomatrix369) [inline makejdk.sh]: GIT_CLONE_ARGUMENTS ==> GIT_FETCH_ARGUMENTS?

File: makejdk.sh [hunk 2]
```diff
@@ -215,9 +215,10 @@ getOpenJDKUpdateAndBuildVersion()
...
+    export OPENJDK_TAG=$(git describe --tags $(git rev-list --tags --max-count=1))
```
Comment (neomatrix369) [inline makejdk.sh]: OPENJDK_TAG ==> OPENJDK_REPO_TAG - more specific to what it contains

File: makejdk.sh (hunk 1 above)
Comment (gdams) [inline makejdk.sh]: no because the same arguments apply right? I assume that's where the shallow setting comes from

**Comment by karianna** [comment]
Will merge after CI passes

**Comment by karianna** [comment]
Hmm build failed

**Comment by gdams** [comment]
for some reason I am not seeing ` echo "${git}Pulling latest tags and getting the latest update version using git fetch -q --tags ${SHALLOW_CLONE_OPTION}"` in the jenkins runs but am when I run it locally... almost like the if statement doesn't work in jenkins

**Comment by gdams** [comment]
CI run https://ci.adoptopenjdk.net/view/OpenJDK%20PR/job/openjdk_pr_pipeline/22/

**Comment by gdams** [comment]
@neomatrix369 / @tellison any idea how to fix https://travis-ci.org/AdoptOpenJDK/openjdk-build/builds/245674305?utm_source=github_status&utm_medium=notification ?

**Comment by neomatrix369** [comment]
@gdams can we check if we are running the same version of `shellcheck` on Travis as on our local machine (and the config we have applied to `shellcheck` on Travis), I'll to look for something quickly for you and come back

File: makejdk.sh [hunk 3]
```diff
@@ -211,15 +212,24 @@ cloneOpenJDKGitRepo()
...
-    echo "${OPENJDK_UPDATE_VERSION} ${OPENJDK_BUILD_NUMBER}"
-    cd "${WORKING_DIR}" || return
+  if [ -d "${WORKING_DIR}/${OPENJDK_REPO_NAME}/.git" ]; then
+    case "${REPOSITORY}" in
+      *openjdk-jdk8u)
+        # It does exist and it's a repo other than the AdoptOpenJDK one
+        cd "${WORKING_DIR}/${OPENJDK_REPO_NAME}" || return
+        echo "${git}Pulling latest tags and getting the latest update version using git fetch -q --tags ${SHALLOW_CLONE_OPTION}"
+        git fetch -q --tags "${SHALLOW_CLONE_OPTION}"
+        export OPENJDK_REPO_TAG="$(git describe --tags $(git rev-list --tags --max-count=1))"
```
Comment (neomatrix369) [inline makejdk.sh]: I think it's better to place this `export OPENJDK_REPO_TAG` on a separate line and then assign it separately here.

Comment (neomatrix369) [inline makejdk.sh]: The way to disable the error would be:
```
THE_FIRST_TAG="$(git rev-list --tags --max-count=1)"
# shellcheck disable=SC2155
export OPENJDK_REPO_TAG="$(git describe --tags $THE_FIRST_TAG)"
```

**Comment by gdams** [comment]
okay @neomatrix369 updated PTAL

**Comment by gdams** [comment]
https://ci.adoptopenjdk.net/view/OpenJDK%20PR/job/openjdk_pr_pipeline/24/

**Comment by karianna** [comment]
Looks like the git tags command is still not taking hold correctly?

**Comment by gdams** [comment]
@karianna it worked on those builds?

``` bash
Pulling latest tags and getting the latest update version using git fetch -q --tags --depth=1
Version: 162 00
```

**Comment by karianna** [comment]
@gdams right you are - the version string is not being set correctly however.  Looking at my original impl to see where it might be going wrong.

**Comment by karianna** [comment]
@gdams - in `configuringVersionStringParameter` can you add the following to the bottom?:

`  echo "Completed configuring the version string parameter, config args are now: ${CONFIGURE_ARGS}"`

**Comment by gdams** [comment]
@karianna updated PTAL

**Comment by karianna** [comment]
```
fatal: No tags can describe '29c2c69bd3c2d03c954bfd54c97cba62edfb8424'.
Try --always, or create some tags.
fatal: No tags can describe '29c2c69bd3c2d03c954bfd54c97cba62edfb8424'.
Try --always, or create some tags.
```

**Comment by karianna** [comment]
I think merging https://github.com/AdoptOpenJDK/openjdk-build/pull/114/ will resolve that

**Comment by gdams** [comment]
@karianna this PR already fixes that issue

**Comment by karianna** [comment]
@gdams I landed 114 so you'll need to rebase and push

**Pull request merged**